backup/bin/setup-barman

```diff
@@ -109,21 +106,15 @@ def initial_backup(conf):
     subprocess.run(["barman", "cron"], check=True)
     for x in conf:
         nm = x['configuration_name']
-        force_archive(nm)
-        print("Running initial backup of {}".format(nm), flush=True)
-        subprocess.run(["barman", "backup", nm], check=True)
-        print("Checking barman status of {}".format(nm), flush=True)
-        subprocess.run(["barman", "check", nm], check=True)
-
-
-def setup_cron(conf):
-    print("Setting up cron jobs", flush=True)
-    # Once-per-minute cron job:
-    cron = "* * * * * root barman cron\n"
-    with open("/etc/cron.d/barman", 'w') as f:
-        f.write(cron)
-    print("Starting cron service", flush=True)
-    subprocess.run(["service", "cron", "start"], check=True)
+        base_dir = '/var/lib/barman/{}/base'.format(nm)
+        if os.path.exists(base_dir) and os.listdir(base_dir):
+            print("Base backup already exists for {}".format(nm))
+        else:
+            force_archive(nm)
+            print("Running initial backup of {}".format(nm), flush=True)
+            subprocess.run(["barman", "backup", nm], check=True)
+            print("Checking barman status of {}".format(nm), flush=True)
+            subprocess.run(["barman", "check", nm], check=True)
```

Comment:
I see, so we only do a base backup if we haven't already done one. And this means we don't need the `no-initial-backup` flag when we call this from the docker entrypoint.

Reply:
Yeah, this way felt simplest to me. It doesn't allow for deploying a barman instance that just streams with no base backup but I don't think that's a useful thing to have really